src/pages/deck-editor/deck-editor.tsx

```diff
@@ -0,0 +1,118 @@
+import { testJapaneseVerbsDeck } from '../../models/mock/deck.mock';
+import { getTestFoxCard, getTestMonkeyCard, getTestMouseCard } from '../../models/mock/card.mock';
+import { AnimatePresence } from 'framer-motion';
+import { Fade } from '../../animations/fade';
+import { v4 as uuidv4 } from 'uuid';
+
+const testDeck = testJapaneseVerbsDeck(0);
+testDeck.cards = [getTestMonkeyCard(), getTestFoxCard(), getTestMouseCard()];
+
+interface DeckEditorPageProps {
+  label?: string;
+  deckId: number;
+  onGoBackClick: (event: React.MouseEvent) => void;
+}
+
+export const DeckEditorPage = ({
+  label = 'edit a deck',
+  deckId,
+  onGoBackClick,
+}: DeckEditorPageProps) => {
+  const [savedDeck, setSavedDeck] = useState(testDeck);
+  const [unsavedDeck, setUnsavedDeck] = useState(testDeck);
+  const isDeckSaved = savedDeck === unsavedDeck;
+
+  return (
+    <div className="deck-editor">
+      <div className="deck-editor-header">
+        <PageHeader label={label} onGoBackClick={onGoBackClick} />
+        {getSaveButtonAndLabel()}
+      </div>
+      <MetaDataEditor
+        deck={unsavedDeck}
+        onDeckChange={handleDeckChange}
+        onDeleteClick={handleDeleteDeckClick}
+      />
+      {unsavedDeck.cards.length > 0 ? getFlashcardSet() : getFlashcardSetPlaceholder()}
+      <Button onClick={handleAddCardClick} className="add-card-button editor-button">
+        <PlusIcon />
+        <label>new card</label>
+      </Button>
+    </div>
+  );
+
+  function getSaveButtonAndLabel() {
+    return (
+      <div>
+        <AnimatePresence>
+          {!isDeckSaved && (
+            <Fade>
+              <Button
+                variant="invisible"
+                onClick={handleDiscardChangesClick}
+                className={`discard-changes-button`}
+              >
+                discard changes
+              </Button>
+            </Fade>
+          )}
+        </AnimatePresence>
+
+        <Button onClick={handleSaveClick} className="editor-button">
+          save
+        </Button>
+      </div>
+    );
+  }
+
+  function getFlashcardSetPlaceholder() {
+    return (
+      <div className="flashcard-set-placeholder">
+        <label>{'you currently have no cards. click "+ new card" to get started'}</label>
+      </div>
+    );
+  }
+
+  function getFlashcardSet() {
+    return (
+      <FlashcardSet
+        cards={unsavedDeck.cards}
+        className="deck-editor-flashcard-set"
+        onCardsChange={(cards: Card[]) => handleDeckChange({ ...unsavedDeck, cards })}
+      />
+    );
+  }
+
+  function handleAddCardClick() {
+    const card = createNewCard(unsavedDeck.frontLang, unsavedDeck.backLang);
+    handleDeckChange({ ...unsavedDeck, cards: [card, ...unsavedDeck.cards] });
+  }
+
+  function handleDeckChange(newDeck: Deck) {
+    setUnsavedDeck(newDeck);
+  }
+
+  function handleSaveClick() {
+    const cards = unsavedDeck.cards.map((card) => ({ ...card, key: uuidv4() }));
```

Comment:
perhaps I'm misunderstanding something here, what's the purpose of remapping to a new key?
just by creating a shallow copy `const newDeck = { ...unsavedDeck }`, we should:
- yield the same behavior since the reference changes `savedDeck === unsavedDeck`
- not have to rerender the entire flashcard set DOM since the keys don't change

Reply:
I was just testing the behavior if we had to reset all the keys after fetching the new cards on save.

 I have it removed now.

Reply:
oh right. yeah we'll still have to see if that's a performance concern with a lot more cards.

something to think about is maybe we should default to using the `id` of the card for the `<Reorder.Item key={...}/>` and fallback to the generated `uuidv4` only if there is no `id` -- that way when we refetch, the keys will be stable for elements that already exist in the DB.

just a thought and will only improve performance on edits / reorders, but still this might be a premature optimization and doesn't matter that much in the end at all

Reply:
Yeah, this will only affect things when they click the save button. I like the idea of using uuid in the id for new cards. 

We would just have to change the card model to be a string for id, then stringify them after we fetch them. 

Would we just see if the id can be parsed as an int to determine if the card is new?

Reply:
Or are you saying to keep the properties separate and only add a key when the card is new?

Reply:
so I was moreso meaning that new cards on the front end should have a `uuid` since they don't exist on the backend yet.
then when we actually post the cards to the backend and refetch, then instead of regenerating a new `uuid` for each card we get back, we just use the `id` of the card from the DB as the key.

this has the advantage that cards that we fetched previously from the backend won't need to be rerendered in the DOM since the `id` will always be the same.

change would be as simple as `<Reorder.Item key={card.id ?? card.key} value={card}>`

Reply:
Yeah, that makes sense. Let's do that! I'll put that in and merge